### PR TITLE
Advance name ccsm cppdefs to cime cppdefs & Update CIME submodule

### PR DESCRIPTION
**Pull request by WesCoomber**
I improved the e3sm commits to
1)Execute new writes to the new renamed CIME_cppdefs file,
2)Open and safe copy the new CIME_cppdefs file, if the file exists
3)Open and safecopy the old CCSM_cppdefs file, if the system can not find the CIME_cppdefs file. 
This supports Rob Jacob's proposal to keep both, CIME_cppdefs and CCSM_cppdefs,
for some time to give models time to adjust.

===================================================================

I also updated the CIME submodule.

EAM and ELM rename changes now on cime master:
   Add EAM case for 20TR CMIP6 docn component
   Allow for different atm names in chem_mech file generation
   Make necessary changes in E3SM cime config files and other files to allow switching cam to eam for E3SM.
   Change name of clm to elm in E3SM

Other changes:
    Advance name of CCSM_cppdefs to CIME_cppdefs
    Add a no-submit option to jenkins_generic_job
    Add "force turning off success update" option to jenkins_generic_job
    In MEMLEAK checks, skip first day memory highwater while initializing.
    Some updates to E3SM test infrastructure
    change permissions on ref case rpointer files
    Add mpi_init timer and ouput memory usage before CIME Run loop
    Add Support csh source cmd
    Add JRA Repeat Year Forcing data modes

Fixes:
    Current mods test, Y_TestUserConcurrentMods.
    Fix check_input_data to fail over to svn when a wget download fails
    Skip preview namelist flag is now honored by case.submit
    Fix mct build issue
    Correctly attribute scorpio timers in cime_pre_init2
    Fix path to cmeps buildexe

No impact on E3SM:
    Fix path to FMS in CESM
    Fix path to cdeps buildlib for cesm
    Update cmeps paths for cesm 
    Add support for MPAS-A grids from @brian-eaton.
    Downgrade Intel Compiler for UFS applications
    Update impi version for ufs model
    Add S2sport

Details:
Updates cime submodule from 54ef8b8b to 7b9ba680af

[BFB]

### CODE REVIEW

File: components/cice/cime_config/buildlib [hunk 1]
```diff
@@ -42,19 +42,33 @@ if (-f "Filepath") {
 #  force gmake to recompile if any of the cppdefs are different 
 #-------------------------------------------------------------- 
 my $recompile = 'FALSE';
-if (-e "$OBJROOT/ice/obj/CCSM_cppdefs") {
+my $sysmod = "cp -f $OBJROOT/ice/obj/CIME_cppdefs.new $OBJROOT/ice/obj/CIME_cppdefs";
```
Comment (jgfouca) [inline components/cice/cime_config/buildlib:45]: There is probably a way to do this without duplicating so much code. Just store the correct cppdef filename root in a variable.

**Review by jgfouca — CHANGES_REQUESTED** [review]
Some minor changes needed. Did you confirm that cases still build OK? Including MPAS?

**Comment by rljacob** [comment]
Wait until https://github.com/ESMCI/cime/pull/3740 is merged and add that to the CIME update.

File: components/cice/cime_config/buildlib [hunk 2]
```diff
@@ -42,19 +42,33 @@ if (-f "Filepath") {
...
+    # the cppdefs are different
+    print "recompile is $recompile \n";
+    if ($recompile eq 'TRUE') {
+        my $sysmod = "rm -f $OBJROOT/ice/obj/*.o";
+        system($sysmod) == 0 or die "ERROR cice buildlib: $sysmod failed: $?\n";
+    }
+    my $sysmod = "cp -f $OBJROOT/ice/obj/CIME_cppdefs.new $OBJROOT/ice/obj/CIME_cppdefs";
+}
+elsif (-e "$OBJROOT/ice/obj/CCSM_cppdefs") {
     if (compare("$OBJROOT/ice/obj/CCSM_cppdefs.new", "$OBJROOT/ice/obj/CCSM_cppdefs") != 0) {
```
Comment (rljacob) [inline components/cice/cime_config/buildlib:60]: You should also replace CCSM_cppdefs.new with CIME_cppdefs.new everywhere.

**Review by rljacob — CHANGES_REQUESTED** [review]

File: components/cice/cime_config/buildlib (hunk 1 above)
Comment (WesCoomber) [reply components/cice/cime_config/buildlib]: Ok  I changed cice/cime_config/buildlib/ to store
the correct cppdef filename root in a variable
which increases code efficiency and reduces
excess duplication of code.

File: components/cice/cime_config/buildlib (hunk 2 above)
Comment (WesCoomber) [reply components/cice/cime_config/buildlib]: Ok I changed cice/cime_config/buildlib/ to store
the correct cppdef filename root in a variable, and
I also replaced CCSM_cppdefs.new with CIME_cppdefs.new everywhere.

**Review by jgfouca — APPROVED** [review]

**Review by rljacob — APPROVED** [review]

**Comment by jgfouca** [comment]
Merging to next

**Pull request merged**